examples/examples_transitive_deps.bzl

```diff
@@ -4,14 +4,37 @@ There are some transitive dependencies of the dependencies of the examples'
 dependencies. This file contains the required macros to pull these dependencies
 """
 
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
 load("@io_bazel_rules_rust//:workspace.bzl", "rust_workspace")
 load("@rules_proto//proto:repositories.bzl", "rules_proto_dependencies", "rules_proto_toolchains")
 
-def transitive_deps():
-    """Define transitive dependencies for `rules_rust` examples"""
+# buildifier: disable=unnamed-macro
+def transitive_deps(is_top_level = False):
+    """Define transitive dependencies for `rules_rust` examples
+
+    Args:
+        is_top_level (bool, optional): Indicates wheather or not this is being called
+            from the root WORKSPACE file of `rules_rust`. Defaults to False.
+    """
 
     rules_proto_dependencies()
 
     rules_proto_toolchains()
 
     rust_workspace()
+
+    # Needed by the hello_uses_cargo_manifest_dir example.
+    if is_top_level:
+        maybe(
```

Comment:
This should be
```python
        maybe(
            native.local_repository,
            name = "hello_cargo_manifest_dir",
            path = "hello_cargo_manifest_dir",
        )
```
(Sorry if this is obvious, just trying to help 😅)

Reply:
"just one small change, surely I can't mess that one up". Anyway, thanks for the help :-)

Unrelated, but I noticed that if you build from the `examples/` dir (which creates all the `examples/bazel-*` directories, building from the root directory sometimes fails because of infinite recursion, as if the `.bazelignore` was ignored. I couldn't find how to fix the issue (the issue wasn't introduced in this commit though).

Reply:
Yeah... that's what happens when you have nested workspaces that depend on the root workspace like this. `local_repository` should ignore bazel output paths (or accept a pattern to ignore certain directories). 

Hopefully [Bazel's proposal for external dependencies](https://docs.google.com/document/d/1moQfNcEIttsk6vYanNKIy3ZuK53hQUFq1b1r0rmsYVg/edit) resolves this. For now, you'll just have to make sure you `bazel clean` in `./examples`. Or at least that's what I do... 😅